### ISSUE TITLE
[Enhancement]: Get aws_eks_cluster and aws_eks_cluster_auth data sources via cluster tags

### ISSUE THREAD

**Issue by dreinhardt89** [hashicorp/terraform-provider-aws#27840]
### Description

We currently use random_pet generation for out cluster names so we don't always know what our cluster names are going to be. It would be helpful if we could get the datasources for ```aws_eks_cluster``` and ```aws_eks_cluster_auth``` via tags.

### Affected Resource(s) and/or Data Source(s)

- aws_eks_cluster
- aws_eks_cluster_auth

### Potential Terraform Configuration

```terraform
data "aws_eks_cluster_auth" "selected" {
  tags = {
    Environment = "Development"
  }
}

data "aws_eks_cluster" "selected" {
  tags = {
    Environment = "Development"
  }
}
```

**Comment by erz4** [comment]
That searching capability can be done via aws console?

**Comment by mattburgess** [comment]
No, it doesn't look like it. `describe-cluster` requires a `--name` parameter and `list-clusters` has no filtering at all beyond the `--include` parameter which enables external clusters to be included alongside Amazon EKS clusters.

@dreinhardt-terminus you might want to consider requesting this feature at https://github.com/aws/containers-roadmap (and linking it to this issue). If you have a support contract with AWS feel free to reach out to your Technical Account Manager as well to see if they can help get eyes on it or at least see if the team already has a similar feature on their roadmap (I couldn't see one after a few searches of that GH repo).
